src/main/java/core/basesyntax/strategy/SupplyOperation.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.Inventory;
+
+public class SupplyOperation implements FruitOperationHandler {
+    @Override
+    public void executeOperation(FruitTransaction transaction, Inventory inventory) {
+        inventory.getInventory().put(transaction.getFruit(),
+                inventory.getInventory().getOrDefault(
+                        transaction.getFruit(), 0) + transaction.getQuantity());
+    }
```

Comment:
It seems like your `SupplyOperation` class directly manipulates the inventory's data structure. According to the SOLID principles, especially the Single Responsibility Principle, it would be better to have a method in the `Inventory` class that encapsulates this logic. This way, `SupplyOperation` doesn't need to know about the internal structure of `Inventory`.